backup-tools/backup-replication-image/go-src/job_queue.go

```
@@ -0,0 +1,269 @@
+
+			for j := range jobs {
+				doWork(i, j)
+			}
+		}(i)
+	}
+
+	for id, file := range EnvFrom.Files {
+
+		if stringInAllSlice(file, EnvTo) {
+			EnvFrom.Files[id] = ""
+			//log.Printf("Skip File: %s its already on all replication regions\n", file)
+		}
+		if strings.HasSuffix(file, "base") {
+			EnvFrom.Files[id] = ""
+		}
+
+	}
+
+	EnvFrom.Files = utils.DeleteEmpty(EnvFrom.Files)
+
+	countAllFiles = len(EnvFrom.Files)
+	PromGauge.AllFiles(countAllFiles)
+
+	// add jobs
+	for _, file := range EnvFrom.Files {
+
+		fileCounter += 1
+
+		jobs <- Job{
+			EnvFrom:      EnvFrom,
+			EnvTo:        EnvTo,
+			File:         file,
+			FileAllCount: countAllFiles,
+			FileNumber:   fileCounter,
+		}
+	}
+	close(jobs)
+
+	// wait for workers to complete
+	wg.Wait()
+
+	os.RemoveAll(backupDir)
+
+	var replicationDuration = time.Since(startTime)
+	log.Printf("End replication task in %v\n", replicationDuration-(replicationDuration%time.Second))
+}
+
+func LoadAndStartJobs() {
+	// cfg used for the parsed YAML Configuration
+	cfg := configuration.YAMLReplication("/backup/env/config.yml")
+
+	// Set all to false for a new loop as default
+	alreadyPrinted = 0
+
+	var err error
+	var tmpExpireInt int
+	tmpExpire := os.Getenv("BACKUP_EXPIRE_AFTER")
+	if tmpExpire == "" {
+		expiration = 864000
+	} else {
+		tmpExpireInt, err = strconv.Atoi(tmpExpire)
+		if err == nil {
+			expiration = int64(tmpExpireInt)
+		} else {
+			expiration = 864000
+		}
+	}
+
+	os.MkdirAll(backupDir, 0777)
+
+	EnvFrom = &Env{Cfg: cfg.From}
+	EnvFrom.Cfg.ContainerPrefix = EnvFrom.Cfg.OsRegionName
+	EnvFrom.SwiftCli = swiftcli.SwiftConnection(
+		EnvFrom.Cfg.OsAuthVersion,
+		EnvFrom.Cfg.OsAuthURL,
+		EnvFrom.Cfg.OsUsername,
+		EnvFrom.Cfg.OsPassword,
+		EnvFrom.Cfg.OsUserDomainName,
+		EnvFrom.Cfg.OsProjectName,
+		EnvFrom.Cfg.OsProjectDomainName,
+		EnvFrom.Cfg.OsRegionName,
+		EnvFrom.Cfg.ContainerPrefix)
+	EnvFrom.Files, _ = swiftcli.SwiftListPrefixFiles(EnvFrom.SwiftCli, EnvFrom.Cfg.ContainerPrefix)
```

Errors should not be swallowed here (also down below).